### PR TITLE
Fix the tuned profile when IRQ load balancing is not disabled globally

### PR DESCRIPTION
**Pull request by marcel-apf**
By default tuned sets the default IRQ SMP affinity to the non isolated
CPUs set.

However, when the IRQ load balancing is set dynamically depending on pod
annotations, the default IRQ SMP affinity should remain all machine
CPUs.

Fix the behavior by setting the new tuned option:
"default_irq_smp_affinity" to "ignore."

Signed-off-by: Marcel Apfelbaum <marcel@redhat.com>

### CODE REVIEW

File: functests/2_performance_update/updating_profile.go
Comment (fromanirh) [inline functests/2_performance_update/updating_profile.go:85]: let's use the full name (`checkDefaultSMPAffinity`) next time

File: functests/2_performance_update/updating_profile.go
Comment (fromanirh) [inline functests/2_performance_update/updating_profile.go:92]: ditto

**Review by fromanirh — APPROVED** [review]
/lgtm
/hold
let's wait for another ack before to merge

File: functests/2_performance_update/updating_profile.go
Comment (marcel-apf) [reply functests/2_performance_update/updating_profile.go:85]: ack

File: functests/2_performance_update/updating_profile.go
Comment (marcel-apf) [reply functests/2_performance_update/updating_profile.go:92]: sure

File: functests/2_performance_update/updating_profile.go
Comment (marcel-apf) [inline functests/2_performance_update/updating_profile.go:38]: @fromanirh please see existing naming.

Comment (fromanirh) [reply functests/2_performance_update/updating_profile.go:38]: sure, let's fix all of them (in another future PR of course).

File: build/assets/tuned/openshift-node-performance
Comment (MarSik) [inline build/assets/tuned/openshift-node-performance:37]: Now looking at this, is our CRI-O logic configuring irqbalance properly when needed?

Comment (marcel-apf) [reply build/assets/tuned/openshift-node-performance:37]: I hop so... What do you mean exactly?

Comment (marcel-apf) [reply build/assets/tuned/openshift-node-performance:37]: @MarSik  When the irqs are not globally disabled we don't ban any CPUs; the pod annotation itself will hint crio to set the default_smp_affinity to all the cpus but this pod's cpu (tested)

Comment (MarSik) [reply build/assets/tuned/openshift-node-performance:37]: But isn't irqbalance a dynamic service that needs to know it should not rebalance (at runtime) to certain cores?

Comment (marcel-apf) [reply build/assets/tuned/openshift-node-performance:37]: cri-o will call irqbalance cli specifying the cores suitable for irqs. This is set is checked against the default_smp_affinity mask.
Before this patch the default_smp_affinity included only the reserved the CPUs, so the call to irqbalance did not do much...

File: functests/2_performance_update/updating_profile.go
Comment (dshchedr) [inline functests/2_performance_update/updating_profile.go:105]: hm, is it typo and these two should be switched?

Comment (marcel-apf) [reply functests/2_performance_update/updating_profile.go:105]: Is not a type, this is what I test:
1. irqLoadBalancingDisabled -> default_smp_affinity should be only  "non-isolated" CPUs
2. irqLoadBalancingEnabled -> default_smp_affinity should be all online cpus

Comment (dshchedr) [reply functests/2_performance_update/updating_profile.go:105]: brr, i'm slightly confused.. 
I'm talking about error message, what I see:

if smpAffinitySet != $onlineCPUS --- show message: "found Default SMP Affinity $onlineCPUS, expected $difference

but I guess it should be: found $difference, expected $onlineCPUS

Comment (dshchedr) [reply functests/2_performance_update/updating_profile.go:105]: the logic itself right, i'm talking about error message only

Comment (marcel-apf) [reply functests/2_performance_update/updating_profile.go:105]: Aha, you are right! Nice catch.

Comment (dshchedr) [reply functests/2_performance_update/updating_profile.go:105]: no any complains other than that, so let's unhold PR when you fix this issue?

File: functests/2_performance_update/updating_profile.go
Comment (marcel-apf) [reply functests/2_performance_update/updating_profile.go:38]: ack

File: functests/2_performance_update/updating_profile.go
Comment (marcel-apf) [reply functests/2_performance_update/updating_profile.go]: done, since the code is re-pushed, can you please lgtm again?

Comment (dshchedr) [reply functests/2_performance_update/updating_profile.go]: /lgtm

**Comment by marcel-apf** [comment]
Addressed @dshchedr review:
 - Fixed wrong error message

File: functests/2_performance_update/updating_profile.go
Comment (marcel-apf) [reply functests/2_performance_update/updating_profile.go]: thanks!!

**Pull request merged**
